stacks/go/devfile.yaml

```diff
@@ -0,0 +1,45 @@
+commands:
+  - exec:
+      commandLine: GOCACHE=/project/.cache go build main.go
+      component: runtime
+      group:
+        isDefault: true
+        kind: build
+      workingDir: /project
+    id: build
+  - exec:
+      commandLine: ./main
+      component: runtime
+      group:
+        isDefault: true
+        kind: run
+      workingDir: /project
+    id: run
+components:
+  - container:
+      endpoints:
+        - name: http
+          targetPort: 8080
+      image: golang:latest
+      memoryLimit: 1024Mi
+      mountSources: true
+      sourceMapping: /project
+    name: runtime
+metadata:
```

Comment:
Nit: Can you move the metadata fields and schema version for each devfile up to the top to be consistent with the other devfiles in this repo? I just find it helps with readability

Reply:
Ill change this with the final commit. Every time I do odo create it changes the formatting :/

Reply:
Oh, that's weird. @yangcao77 @maysunfaisal is this the issue with the yaml parser we've brought up before, or something else with the devfile library?

Reply:
yep, this is the yaml/json encoding issue we've seen before. However, pure `odo create` should not mess up the formatting. any action to update/modify the devfile through odo would result in this issue, since it re-writes the devfile.

Reply:
Maybe I am confusing it when I do `odo create url`. In my next commit I'll fix this.